### ISSUE TITLE
The db_remove() function from salt.modules.mysql mistakenly tests the name against "information_scheme"

### ISSUE THREAD

**Issue by arizvisa** [saltstack/salt#54938]
### Description of Issue
This is a pretty insignificant bug, but in the `salt.modules.mysql` module, there's a misspelling when checking to see if the provided database name is invalid.

https://github.com/saltstack/salt/blob/master/salt/modules/mysql.py#L1184

```
    if name in ('mysql', 'information_scheme'):
        log.info('DB \'%s\' may not be removed', name)
        return False
```

When calling `db_remove()`, the function first checks if the named database exists and then checks the name against a tuple. One of these names, "information_scheme" is misspelled and should be "information_schema". Thus, this results in different semantics than from what the author intended.

I'm pretty sure that you're not able to drop the "information_schema" table though, and it's pretty crazy to try as it's used for reflection. So it's a 1-byte fix.

### Setup
This is in the `salt/modules/mysql.py` file from the master branch.

### Steps to Reproduce Issue
Probably call db.remove like:
```
salt \* mysql.db_remove information_schema
```

### Versions Report
This was just found while doing something else in the master branch.

**Comment by arizvisa** [comment]
This is literally a 1-character patch and shouldn't take long to fix. The testcase for it is pretty simple too.

*snaps whip at bot*

**Comment by Ch3LL** [comment]
thanks for pointing this one out. Any chance you want to give it a go and push a PR?

**Comment by arizvisa** [comment]
Yeah, I can. I'm still waiting for a couple of my PRs to get merged/ported into master though. So I'm hesitant to submit anything until those make it through.

**Comment by arizvisa** [comment]
Specifically #55186
